sdcm/scan_operation_thread.py

```diff
@@ -0,0 +1,352 @@
+        self.table_clustering_order = self.get_table_clustering_order()
+        self.reversed_order = 'desc' if self.table_clustering_order.lower() == 'asc' else 'asc'
+        self.reversed_query_filter_ck_stats = {'lt': {'count': 0, 'total_scan_duration': 0},
+                                               'gt': {'count': 0, 'total_scan_duration': 0},
+                                               'lt_and_gt': {'count': 0, 'total_scan_duration': 0},
+                                               'no_filter': {'count': 0, 'total_scan_duration': 0}}
+        self.ck_filter = ''
+
+    def get_table_clustering_order(self) -> str:
+        for node in self.db_cluster.nodes:
+            try:
+                with self.create_session(node) as session:
+                    return get_table_clustering_order(ks_cf=self.ks_cf, ck_name=self.ck_name, session=session)
+            except Exception as error:  # pylint: disable=broad-except
+                self.log.info('Failed getting table %s clustering order through node %s : %s', self.ks_cf, node.name,
+                              error)
+        raise Exception('Failed getting table clustering order from all db nodes')
+
+    def randomly_form_cql_statement(self) -> Optional[tuple[str, str]]:  # pylint: disable=too-many-branches
+        """
+        The purpose of this method is to form a random reversed-query out of all options, like:
+            select * from scylla_bench.test where pk = 1234 and ck < 4721 and ck > 2549 order by ck desc
+
+        This consists of:
+        1) Select a random partition found in requested table.
+        2) randomly add a CQL LIMIT and 'bypass cache' to query.
+        3) Add a random CK filter with random row values.
+        :return: a CQL reversed-query
+        """
+        with self.create_session(self.db_node) as session:
+            ck_name = self.ck_name
+            rows_count = self.rows_count
+            ck_random_min_value = random.randint(a=1, b=rows_count)
+            ck_random_max_value = random.randint(a=ck_random_min_value, b=rows_count)
+            self.ck_filter = ck_filter = random.choice(list(self.reversed_query_filter_ck_by.keys()))
+            pk_name = self.pk_name
+            if pks := get_partition_keys(ks_cf=self.ks_cf, session=session, pk_name=pk_name):
+                partition_key = random.choice(pks)
+                # Form a random query out of all options, like:
+                # select * from scylla_bench.test where pk = 1234 and ck < 4721 and ck > 2549 order by ck desc
+                # limit 3467 bypass cache
+                normal_query = reversed_query = self.basic_query.format(
+                    self.ks_cf) + f' where {pk_name} = {partition_key}'
+                query_suffix = limit = ''
+
+                if random.choice([False] + [True]):  # Randomly add a LIMIT
+                    limit = random.randint(a=rows_count/5, b=rows_count)
```

Comment:
Why 5? just something you picked?
maybe should be picked randomly?

Reply:
ok changed to "a=1" so it is fully random and can get very small numbers.